File: practice/algorithms/bit_manipulation/maximizing_xor/maximizing_xor.cpp
#include <iostream>

using namespace std;

int max_xor(int i_l, int i_r)
{
    int maxv = 0;
    for (int i = i_l; i < i_r + 1; ++i)
    {
        for (int j = i_l; j < i; ++j)
        {
            int x = i ^ j;
            if (x > maxv)
            {
                maxv = x;
            }
        }
    }
    
    return maxv;
}

int main()
{
    int res, l, r;
    cin >> l >> r;
    res = max_xor(l, r);
    cout << res << endl;
    
    return 0;
}

File: practice/algorithms/bit_manipulation/maximizing_xor/maximizing_xor.py
#!/usr/bin/env python

def max_xor(i_l, i_r):
    maxv = 0
    for i in xrange(i_l, i_r + 1):
        for j in xrange(i_l, i):
            x = i ^ j
            if x > maxv:
                maxv = x
    return maxv

l = int(raw_input().strip())
r = int(raw_input().strip())
print max_xor(l, r)

